app/controllers/links_controller.rb

```diff
@@ -0,0 +1,40 @@
+class LinksController < ApplicationController
```

Comment:
Is this a controller instead of a service because of Javascript?

Reply:
Yes, the idea is that once we have this controller, we can generate links on the server, classic JS, angular or react. All of them will go through a single controller that will be responsible for bookkeeping of all the external links.